### PR TITLE
Fixing vendor-specific/tied memory property flag detection

### PR DESCRIPTION
**Pull request by toomuchvoltage**
Fixing vendor-specific/tied memory property flag detection for `vkloadtests`'s VMA wrapper.

### CODE REVIEW

**Comment by toomuchvoltage** [comment]
Dear @MarkCallow , this should fix the reported issue. Though would appreciate a quick smoke test on the M2 device as well just to be sure.

**Comment by MarkCallow** [comment]
> Though would appreciate a quick smoke test on the M2 device

It works.

File: tests/loadtests/vkloadtests/VulkanLoadTestSample.cpp [hunk 1]
```diff
@@ -75,7 +78,8 @@ namespace VMA_CALLBACKS
     {
         uint64_t allocId = mt64();
         VmaAllocationCreateInfo pCreateInfo = {};
-        if (allocInfo->memoryTypeIndex == 8)
+        if ((cachedDevMemProps.memoryTypes[allocInfo->memoryTypeIndex].propertyFlags & VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT) ||
+            (cachedDevMemProps.memoryTypes[allocInfo->memoryTypeIndex].propertyFlags & VK_MEMORY_PROPERTY_HOST_COHERENT_BIT))
```
Comment (MarkCallow) [inline tests/loadtests/vkloadtests/VulkanLoadTestSample.cpp:82]: Leave these 2 alone as splitting the &s' operands will make it too hard to follow the logic.

**Review by MarkCallow — CHANGES_REQUESTED** [review]

**Comment by MarkCallow** [comment]
For orthogonality please add `ktxTexture[12]_VkUploadEx_WithSuballocator wrappers to vkloader.c. I hadn't noticed they weren't there in the previous PR.

**Comment by toomuchvoltage** [comment]
> For orthogonality please add `ktxTexture[12]_VkUploadEx_WithSuballocator wrappers to vkloader.c. I hadn't noticed they weren't there in the previous PR.

All done! Let me know if this suffices. 🙏

**Review by MarkCallow — APPROVED** [review]

**Comment by MarkCallow** [comment]
Thanks again.

**Comment by toomuchvoltage** [comment]
> Thanks again.

You're very welcome Mark. I feel privileged to have been able to contribute.
